### PR TITLE
*: TIDB_DECODE_KEY(): Decode keys which only have a TableID

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

The `TIDB_DECODE_KEY()` function was only able to decode keys that were "record" or "index" keys. However the `START_KEY` and `END_KEY` columns of `information_schema.TIKV_REGION_STATUS` always have entries that only have a "table_id" and no row/index info. This is for the minimum start key and the maximum end key.

This change:

- Helps to decode entries that only have an table_id like `7480000000000000FF4700000000000000F8`
- Removes warnings for calling `TIDB_DECODE_ROW()` on the minimum `START_KEY` and maximum `END_KEY`.
- Removes invalid JSON values that are returned by `TIDB_DECODE_ROW()` for the minimum `START_KEY` and maximum `END_KEY`. This allows one to use JSON functions directly without having to use `IF(JSON_VALID(),...`

This is related to https://github.com/pingcap/docs/pull/6728

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

```sql
sql> SELECT TIDB_DECODE_KEY('7480000000000000FF4700000000000000F8');
+---------------------------------------------------------+
| TIDB_DECODE_KEY('7480000000000000FF4700000000000000F8') |
+---------------------------------------------------------+
| {"table_id":71}                                         |
+---------------------------------------------------------+
1 row in set (0.0013 sec)
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
TIDB_DECODE_KEY() now works on keys that only have a Table ID
```

### CODE REVIEW

File: planner/core/expression_rewriter.go [hunk 1]
```diff
@@ -2150,6 +2157,16 @@ func decodeIndexKey(key []byte, tableID int64, tbl table.Table, loc *time.Locati
 	return string(retStr), nil
 }
 
+func decodeTableKey(key []byte, tableID int64, tbl table.Table, loc *time.Location) (string, error) {
+	ret := make(map[string]interface{})
+	ret["table_id"] = tableID
```
Comment (kennytm) [inline planner/core/expression_rewriter.go:2162]:
```suggestion
	ret := map[string]int64{"table_id": tableID}
```

or perhaps even 

```go
return fmt.Sprintf(`{"table_id":%d}`, tableId)
```

**Review by tangenta — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Comment by kennytm** [comment]
/run-check_dev_2

:thinking: https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/43863/pipeline

```
[2021-11-07T15:38:01.515Z] --- FAIL: TestCacheTableBasicScan (3.75s)
[2021-11-07T15:38:01.515Z]     result.go:50: 
[2021-11-07T15:38:01.515Z]         	Error Trace:	result.go:50
[2021-11-07T15:38:01.515Z]         	            				cache_test.go:79
[2021-11-07T15:38:01.515Z]         	            				cache_test.go:113
[2021-11-07T15:38:01.515Z]         	Error:      	Not equal: 
[2021-11-07T15:38:01.515Z]         	            	expected: "[UnionScan 3333.33 root  gt(test.tmp1.id, 4)]\n[└─TableReader 3333.33 root  data:TableRangeScan]\n[  └─TableRangeScan 3333.33 cop[tikv] table:tmp1 range:(4,+inf], keep order:true, stats:pseudo]\n"
[2021-11-07T15:38:01.515Z]         	            	actual  : "[TableReader 3333.33 root  data:TableRangeScan]\n[└─TableRangeScan 3333.33 cop[tikv] table:tmp1 range:(4,+inf], keep order:true, stats:pseudo]\n"
[2021-11-07T15:38:01.515Z]         	            	
[2021-11-07T15:38:01.515Z]         	            	Diff:
[2021-11-07T15:38:01.515Z]         	            	--- Expected
[2021-11-07T15:38:01.515Z]         	            	+++ Actual
[2021-11-07T15:38:01.515Z]         	            	@@ -1,4 +1,3 @@
[2021-11-07T15:38:01.515Z]         	            	-[UnionScan 3333.33 root  gt(test.tmp1.id, 4)]
[2021-11-07T15:38:01.515Z]         	            	-[└─TableReader 3333.33 root  data:TableRangeScan]
[2021-11-07T15:38:01.515Z]         	            	-[  └─TableRangeScan 3333.33 cop[tikv] table:tmp1 range:(4,+inf], keep order:true, stats:pseudo]
[2021-11-07T15:38:01.515Z]         	            	+[TableReader 3333.33 root  data:TableRangeScan]
[2021-11-07T15:38:01.515Z]         	            	+[└─TableRangeScan 3333.33 cop[tikv] table:tmp1 range:(4,+inf], keep order:true, stats:pseudo]
```

**Pull request merged**